slides/intro-extra-recipes.qmd

```diff
@@ -0,0 +1,378 @@
+
+*Remember that the transformations are tied to the specific modeling problem.*
+
+```{r}
+#| label: datetime-split-exercise
+#| echo: false
+countdown(minutes = 3, id = "datetime-split")
+```
+
+## Two types of transformations
+
+::: columns
+::: {.column width="50%"}
+
+### Static
+
+- Square root, log, inverse
+- Dummies for known levels
+- Date time extractions
+
+:::
+
+::: {.column width="50%"}
+
+### Trained
+
+- Centering & scaling
+- Imputation
+- PCA
+- Anything for unknown factor levels
+
+:::
+
+:::
+
+::: fragment
+Trained methods needs to calculate **sufficient information** to be applied again
+:::
+
+## The recipes package
+
+::: {.incremental .highlight-last}
+- Modular + extensible
+- Works well with pipes ,`|>` and `%>%`
+- Deferred evaluation
+- Isolates test data from training data
+- Can do things formulas can't
+:::
+
+
+## What is a recipe?
+
+:::{style="font-family: 'Source Code Pro', monospace; font-size: 0.8em;"}
+recipe(tip ~ ., data = taxi_train) %>%  
+\ \ step_unknown(all_nominal_predictors()) %>%  
+\ \ step_dummy(all_nominal_predictors()) %>%  
+\ \ step_zv(all_predictors()) %>%  
+\ \ step_log(distance) %>%  
+\ \ step_normalize(all_numeric_predictors())
+:::
+
+## What is a recipe?
+
+:::{style="font-family: 'Source Code Pro', monospace; font-size: 0.8em;"}
+[recipe(tip ~ ., data = taxi_train)]{style="color: #CA225E;"} %>%  
+\ \ step_unknown(all_nominal_predictors()) %>%  
+\ \ step_dummy(all_nominal_predictors()) %>%  
+\ \ step_zv(all_predictors()) %>%  
+\ \ step_log(distance) %>%  
+\ \ step_normalize(all_numeric_predictors())
+:::
+
+<br>
+
+Start by calling `recipe()` to denote the data source and variables used
+
+## What is a recipe?
+
+:::{style="font-family: 'Source Code Pro', monospace; font-size: 0.8em;"}
+recipe(tip ~ ., data = taxi_train) %>%  
+\ \ [step_unknown]{style="color: #CA225E;"}(all_nominal_predictors()) %>%  
+\ \ [step_dummy]{style="color: #CA225E;"}(all_nominal_predictors()) %>%  
+\ \ [step_zv]{style="color: #CA225E;"}(all_predictors()) %>%  
+\ \ [step_log]{style="color: #CA225E;"}(distance) %>%  
+\ \ [step_normalize]{style="color: #CA225E;"}(all_numeric_predictors())
+:::
+
+<br>
+
+Specify what actions to take by adding `step_*()`s.
+
+## What is a recipe?
+
+:::{style="font-family: 'Source Code Pro', monospace; font-size: 0.8em;"}
+recipe(tip ~ ., data = taxi_train) %>%  
+\ \ step_unknown([all_nominal_predictors()]{style="color: #CA225E;"}) %>%  
+\ \ step_dummy([all_nominal_predictors()]{style="color: #CA225E;"}) %>%  
+\ \ step_zv([all_predictors()]{style="color: #CA225E;"}) %>%  
+\ \ step_log([distance]{style="color: #CA225E;"}) %>% 
+\ \ step_normalize([all_numeric_predictors()]{style="color: #CA225E;"})
+:::
+<br>
+
+Use {tidyselect} and recipes-specific selectors to denote affected variables.
+
+## Using a recipe
+
+:::{style="font-family: 'Source Code Pro', monospace; font-size: 0.8em;"}
+rec_spec <- recipe(tip ~ ., data = taxi_train) %>%  
```

Comment:
Because of the change in title, indentation in first line of code, and bullet point beneath, it was difficult to tell what I should be paying attention to when this slides pops up.

Would it be possible to set indentation in the previous slides s.t. the existing code doesn't move, and the only addition is the assignment `rec_spec <-`? Or highlight `rec_spec <-` in the same way other components have been highlighted? Or assign `rec_spec <-` throughout all of the previous slides, since we don't print out the object or pipe in-line?

Reply:
I like the idea of always doing `rec_spec <-` so I did that in [`34f66c7` (#233)](https://github.com/tidymodels/workshops/pull/233/commits/34f66c72fb2b704e7600ed8194f329532db35102)